rasa/shared/nlu/training_data/util.py

```diff
@@ -210,3 +211,16 @@ def build_entity(
 
     entity.update(kwargs)
     return entity
+
+
+def sparse_matrix_to_string(m: scipy.sparse.spmatrix) -> Text:
+    """Turn sparse matrix into string.
+
+    taken from official scipy source to operate on full sparse matrix.
+    https://github.com/scipy/scipy/blob/v1.7.0/scipy/sparse/base.py#L258
+    Normally you would have to 1) change the maxprint property in-place first,
```

Comment:
could we copy, and then change maxprint and str() the copy?

Reply:
Yes we could - but is it much better than the current solution?

Reply:
I just wonder if this is stable or reliant too much on internal implementation. Is the "normal" way you explained documented anywhere?

Reply:
the one I explained in the comment isn't the normal way, just one way. Yours is another one. And the one implemented yet another one. Hard to tell what would be more stable - i mean potentially they could change the `maxprint` stuff too and we'd have to adjust. Just feel that the current implemented version here is the lesser evil :D

Reply:
What about we call `spmatrix.toarray` and then just use the same thing we did for `ndarray`?

Reply:
Yet another option, but it's probably the most costly in terms of cpu and memory. In the worst case you are turning a handful of numbers into 'a million zeroes + those numbers' just to print them. Also calculating the hash will take longer on a longer string.

Reply:
ok 👍 
Could you add a small comment explaining how the `tocoo` approach works?